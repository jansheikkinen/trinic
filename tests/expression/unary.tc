// Unary

function main(void) -> void do
  ~1; !1; not 1; -1; *1; &1;
end
