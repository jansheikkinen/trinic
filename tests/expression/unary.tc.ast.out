
### AST GENERATION ###
Declaration FUNCTION  main
  Stmt EXPRESSION
    Expr UNARY ~
      Expr LITERAL
        INT 1
  Stmt EXPRESSION
    Expr UNARY !
      Expr LITERAL
        INT 1
  Stmt EXPRESSION
    Expr UNARY NOT
      Expr LITERAL
        INT 1
  Stmt EXPRESSION
    Expr UNARY -
      Expr LITERAL
        INT 1
  Stmt EXPRESSION
    Expr UNARY *
      Expr LITERAL
        INT 1
  Stmt EXPRESSION
    Expr UNARY &
      Expr LITERAL
        INT 1
